eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -1249,6 +1249,54 @@ public function testLoadLocationChildrenDataWithOffsetAndLimit(LocationList $loc
         );
     }
 
+    /**
+     * @covers \eZ\Publish\API\Repository\LocationService::loadLocationChildren
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testLoadLocationChildrenRespectsParentSortingClauses(): void
+    {
+        $repository = $this->getRepository();
+        $locationService = $repository->getLocationService();
+
+        $parentLocation = $locationService->loadLocation(5);
+
+        // Update Location in order to change sort clauses to Name DESC
+        $locationUpdateStruct = $locationService->newLocationUpdateStruct();
+        $locationUpdateStruct->sortField = Location::SORT_FIELD_NAME;
+        $locationUpdateStruct->sortOrder = Location::SORT_ORDER_DESC;
+        $parentLocation = $locationService->updateLocation(
+            $parentLocation,
+            $locationUpdateStruct
+        );
+
+        $children = $locationService->loadLocationChildren($parentLocation);
+        $namesWithDefaultSorting = [];
+        foreach ($children as $child) {
+            $namesWithDefaultSorting[] = $child->getContentInfo()->name;
+        }
+
+        // Update Location in order to change sort clauses to NAME ASC
+        $locationUpdateStruct = $locationService->newLocationUpdateStruct();
+        $locationUpdateStruct->sortField = Location::SORT_FIELD_NAME;
```

Comment:
We should rather have a data provider-based test covering `Location::SORT_FIELD_*` cases.